ldap_auth_provider.py

```diff
@@ -570,29 +578,43 @@ def _ldap_authenticated_search(self, server, password, filters):
             logger.warning("Error during LDAP authentication: %s", e)
             raise
 
-    def map_login_to_upn(self, username):
+    def _map_login_to_upn(self, username):
+        """Maps user provided login to Active Directory UPN and
+        local part of Matrix ID.
+
+        Args:
+            username (str): The user's login
+
+        Raises:
+            ActiveDirectoryUPNException: if username can not be
+                mapped to userPrincipalName
+
+        Returns:
+            Tuple[str, str, str]: a tuple of Active Directory login,
+            Active Directory domain and local part of Matrix ID.
+        """
         login = username.lower()
         domain = self.ldap_default_domain
         localpart = username
 
         if '\\' in username:
             (domain, login) = username.lower().rsplit('\\', 1)
-        elif self.ldap_separator in username:
-            (login, domain) = username.lower().rsplit(self.ldap_separator, 1)
+        elif "/" in username:
+            (login, domain) = username.lower().rsplit("/", 1)
         else:
             if not self.ldap_default_domain:
-                logger.debug(
+                logger.info(
                     'No LDAP separator %s was found in uid %s '
                     'and LDAP default domain was not configured. ',
-                    self.ldap_separator,
+                    "/",
                     self.ldap_default_domain
                 )
                 raise ActiveDirectoryUPNException()
```

Comment:
seems like it might be simpler to just `return None` here, but :man_shrugging:

Reply:
If we will use ```None``` for that, then we can not just deserialize method output like

```
(login, domain, localpart) = self._map_login_to_upn(username)
```

So we will need some temporal variable like:
```
map_array = self._map_login_to_upn(username)
if not map_array:
    error here
(login, domain, localpart) = map_array
```

If you think it's better suits this project then I will refactor code. But I believe exception is better choice here.

Reply:
fair enough.